app/src/main/java/com/babylon/wallet/android/presentation/wallet/WalletViewModel.kt

```diff
@@ -110,7 +112,20 @@ class WalletViewModel @Inject constructor(
         observeProfileBackupState(getBackupStateUseCase)
         observeGlobalAppEvents()
         loadAssets(withRefresh = false)
-        observeNpsSurveyState()
+        observePromptMessageStates()
+    }
+
+    private fun observePromptMessageStates() {
+        viewModelScope.launch {
+            p2PLinksRepository.showRelinkConnectors()
+                .collect { showRelinkConnectors ->
+                    if (showRelinkConnectors) {
```

Comment:
have you double checked that when there is a re-link message shown, and we stay in app and do transaction that fires NPS, it is correctly handled?

Reply:
I've tested by hardcoding the npsSurveyState and it works fine. Please let me know if there might be differences in a real scenario and how I can test it.